internal/provider/auth_azure.go

```diff
@@ -0,0 +1,189 @@
+					"authentication scenario.",
+				ConflictsWith: []string{fmt.Sprintf("%s.0.%s", authField, consts.FieldJWT)},
+			},
+			consts.FieldClientID: {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Description:   "The identity's client ID.",
+				ConflictsWith: []string{fmt.Sprintf("%s.0.%s", authField, consts.FieldJWT)},
+			},
+			consts.FieldScope: {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Default:       defaultAzureScope,
+				Description:   "The scopes to include in the token request.",
+				ConflictsWith: []string{fmt.Sprintf("%s.0.%s", authField, consts.FieldJWT)},
+			},
+		},
+	}, consts.MountTypeAzure)
+}
+
+type AuthLoginAzure struct {
+	AuthLoginCommon
+}
+
+// MountPath for the azure authentication engine.
+func (l *AuthLoginAzure) MountPath() string {
+	if l.mount == "" {
+		return l.Method()
+	}
+	return l.mount
+}
+
+// LoginPath for the azure authentication engine.
+func (l *AuthLoginAzure) LoginPath() string {
+	return fmt.Sprintf("auth/%s/login", l.MountPath())
+}
+
+func (l *AuthLoginAzure) Init(d *schema.ResourceData, authField string) error {
+	if err := l.AuthLoginCommon.Init(d, authField); err != nil {
+		return err
+	}
+
+	if err := l.checkRequiredFields(d, l.requiredParams()...); err != nil {
+		return err
+	}
+
+	if err := l.checkFieldsOneOf(d, consts.FieldVMName, consts.FieldVMSSName); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (l *AuthLoginAzure) requiredParams() []string {
+	return []string{consts.FieldRole, consts.FieldSubscriptionID, consts.FieldResourceGroupName}
+}
+
+// Method name for the azure authentication engine.
+func (l *AuthLoginAzure) Method() string {
+	return consts.AuthMethodAzure
+}
+
+// Login using the azure authentication engine.
+func (l *AuthLoginAzure) Login(client *api.Client) (*api.Secret, error) {
+	if !l.initialized {
+		return nil, fmt.Errorf("auth login not initialized")
+	}
+
+	params, err := l.copyParams(l.requiredParams()...)
+	if err != nil {
+		return nil, err
+	}
+
+	if v, ok := l.params[consts.FieldVMName]; ok {
+		params[consts.FieldVMName] = v
+	} else if v, ok := l.params[consts.FieldVMName]; ok {
+		params[consts.FieldVMSSName] = v
+	}
+
+	ctx := context.Background()
+	jwt, err := l.getJWT(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	params[consts.FieldJWT] = jwt
+	return l.login(client, l.LoginPath(), params)
+}
+
+func (l *AuthLoginAzure) getJWT(ctx context.Context) (string, error) {
+	if v, ok := l.params[consts.FieldJWT]; ok {
+		return v.(string), nil
+	}
+
+	// attempt to get the token from Azure
+	credOpts := &azidentity.ManagedIdentityCredentialOptions{}
+	if v, ok := l.params[consts.FieldClientID]; ok {
+		credOpts.ID = azidentity.ClientID(v.(string))
+	}
+
+	creds, err := azidentity.NewManagedIdentityCredential(credOpts)
+	if err != nil {
+		return "", err
+	}
+
+	tOpts := policy.TokenRequestOptions{
+		Scopes: []string{"https://management.azure.com/"},
```

Comment:
```suggestion
		Scopes: []string{defaultAzureScope},
```

Reply:
Fixed in 2855a9dc194b4b3cf2caf0935b8997265b78f979

Reply:
Ah nice!